### DISCUSSION TITLE
Serverless functions not working properly

### DISCUSSION THREAD

**Discussion by javierfuentesm** [vercel/community#5619] [Help] [answered]
### Page to Investigate

https://globalhitss.vercel.app/signin

### Steps to Reproduce

I didn't touch my deployment in the last 43 days and suddenly everything stopped working, and many requests were failing, I noticed that it was because of the authentication so I thought maybe I just needed to refresh the session, and I logged out and for my surprise, I couldn't log in (I use next-auth) , so I tried redeploying, it didn't work, I push something directly to main and for my surprise, it didn't trigger a build, so I had to create a PR and merge it in order to "force" a deploy  but now is not working the logs show as if everything was fine but I can not pass the login  
<img width="877" alt="image" src="https://github.com/vercel/community/assets/19244527/6cbb4430-4c54-4f41-bb45-3233ac1509e4">

Here is a video of how it is behaving, also it is worth to mention that everything works locally and everything was working fine until yesterday night 

https://github.com/vercel/community/assets/19244527/2145e77b-ea2d-4eb4-ae69-2b2b37d57528

**Comment by rphi** [comment, marked as answer]
Hi Javier - in case it's any use to you, we had a very similar (identical) issue on our app where server-side rendering nextauth sessions just weren't working. In our case, we had the `NEXTAUTH_URL` env variable set in our Vercel environment. Hasn't caused us any issues since we launched, but seems a few days ago Vercel changed something that broke that behaviour.

TL;DR: if you have the `NEXTAUTH_URL` env variable set and you're on Vercel, try removing it and redeploying. That fixed the issue for us.

**Reply by javierfuentesm** [reply]
you are a life saver!!! that worked , do you know why this happen?

**Reply by mobiustripper42** [reply]
THANK YOU!!!
